emily/handler/src/api/handlers/internal.rs

```diff
@@ -0,0 +1,220 @@
+                info!("Successfully set api state: {:?}.", api_state);
+                break;
+            }
+            // Retry if there was a version conflict.
+            Err(Error::VersionConflict) => {
+                if update_attempts >= MAX_SET_API_STATE_ATTEMPTS_DURING_REORG {
+                    debug!("Failed to update API state {:?}", api_state);
+                    return Err(Error::InternalServer);
+                } else {
+                    debug!("Failed to update API state - retrying: {:?}", api_state);
+                }
+            }
+            // If some other error occured then return from here; this shouldn't
+            // happen and something has actually gone wrong.
+            Err(e) => Err(e)?,
+        }
+    }
+    // Return.
+    Ok(Some(api_state))
+}
+
+/// Handler that executes a reorg.
+///
+/// This function isn't intendeded to be exposed into any specific endpoint
+/// outside of what could maybe be a testing endpoint one day. It handles
+/// the internal requests to execute a reorg.
+pub async fn execute_reorg_handler(
+    context: &EmilyContext,
+    request: ExecuteReorgRequest,
+) -> Result<impl warp::reply::Reply, Error> {
+    info!("Executing a reorg with request {request:?}.");
+    let empty_reply = warp::reply::with_status(warp::reply(), StatusCode::NO_CONTENT);
+
+    let new_status = ApiStatus::Reorg(request.canonical_tip.clone().into());
+    match set_api_state_status(context, &new_status).await {
+        // Do nothing if we claimed the api correctly.
+        Ok(Some(_)) => {}
+        Ok(None) => {
+            return Ok(empty_reply);
+        }
+        Err(e) => {
+            return Err(e);
+        }
+    };
+
+    // We have control of the API at this point. For each entry of the deposit
+    // and withdrawal table we'll wipe out all the history that's no longer relevant.
+
+    // Get all deposits that would be impacted by this reorg.
+    let all_deposits = accessors::get_all_deposit_entries_modified_after_height(
+        context,
+        request.canonical_tip.stacks_block_height,
+        None,
+    )
+    .await?;
+
+    // Setup debug modified deposit list.
+    let mut debug_modified_deposit_entries: Vec<DepositEntry> =
+        Vec::with_capacity(all_deposits.len());
+
+    // Kill the history from all the deposits.
+    for deposit in all_deposits {
+        for attempt in 0..ENTRY_UPDATE_RETRIES {
+            let mut entry =
+                accessors::get_deposit_entry(context, &deposit.primary_index_key).await?;
+            entry.reorganize_around(&request.canonical_tip)?;
+            match accessors::set_deposit_entry(context, &mut entry).await {
+                Ok(_) => break,
+                Err(Error::VersionConflict) => {
+                    debug!(
+                        "Encountered race condition in updating entry {:?}. Attempt {}/{}",
+                        entry, attempt, ENTRY_UPDATE_RETRIES
+                    );
+                }
```

Comment:
Similar answer as before; this failed at updating an entry but successfully completing this isn't gatekept by a different process completing.

This only occurs if, between the lines `150` and `153`, another lambda manages to update the deposit.

This should actually be impossible at this moment because at this point the api status was successfully updated and all api calls that can update the deposits and withdrawals will temporarily all fail; but this is here for redundancy.